base/common/src/main/java/com/netscape/certsrv/profile/ProfileData.java

```diff
@@ -273,18 +286,339 @@ public boolean equals(Object obj) {
+            pdElement.appendChild(enabledByElement);
+        }
+        if (authenticatorId != null) {
+            Element authenticatorIdElement = document.createElement("authenticatorId");
+            authenticatorIdElement.appendChild(document.createTextNode(authenticatorId));
+            pdElement.appendChild(authenticatorIdElement);
+        }
+        if (authzAcl != null) {
+            Element authzAclElement = document.createElement("authzAcl");
+            authzAclElement.appendChild(document.createTextNode(authzAcl));
+            pdElement.appendChild(authzAclElement);
+        }
+        Element renewalElement = document.createElement("renewal");
+        renewalElement.appendChild(document.createTextNode(Boolean.toString(renewal)));
+        pdElement.appendChild(renewalElement);
+
+        Element xmlOutputElement = document.createElement("xmlOutput");
+        xmlOutputElement.appendChild(document.createTextNode(Boolean.toString(xmlOutput)));
+        pdElement.appendChild(xmlOutputElement);
+        for (ProfileInput pi : inputs) {
+            Element piElement = pi.toDOM(document);
+            pdElement.appendChild(piElement);
+        }
+        for (ProfileOutput po : outputs) {
+            Element poElement = po.toDOM(document, "Output");
+            pdElement.appendChild(poElement);
+        }
+        if (!policySets.isEmpty()) {
+            Element policysetsElement = document.createElement("PolicySets");
+            for (Entry<String, List<ProfilePolicy>> policySet : getPolicySets().entrySet()) {
+                Element policysetElement = document.createElement("PolicySet");
+                Element idElement = document.createElement("id");
+                idElement.appendChild(document.createTextNode(policySet.getKey()));
+                policysetElement.appendChild(idElement);
+                Element valueElement = document.createElement("value");
+                for (ProfilePolicy pp : policySet.getValue()) {
+                    valueElement.setAttribute("id", pp.getId());
+                    if (pp.getDef() != null) {
+                        Element defElement = document.createElement("def");
+                        defElement.setAttribute("id", pp.getDef().getName());
+                        defElement.setAttribute("classId", pp.getDef().getClassId());
+                        valueElement.appendChild(defElement);
+                        if (pp.getDef().getText() != null) {
+                            Element descriptionElement = document.createElement("description");
+                            descriptionElement.appendChild(document.createTextNode(pp.getDef().getText()));
+                            defElement.appendChild(descriptionElement);
+                        }
+                        if (!pp.getDef().getAttributes().isEmpty()) {
+                            for (ProfileAttribute attribute : pp.getDef().getAttributes()) {
```

Comment:
I think the `if-statement` is not necessary since we will not enter the loop if there are no attributes.

Reply:
Agreed, fixed in next revision.